xdsl/dialects/memref.py

```diff
@@ -185,6 +200,67 @@ def verify_(self) -> None:
                 "op dimension operand count does not equal memref dynamic dimension count."
             )
 
+    def print(self, printer: Printer):
+        printer.print_string("(")
+        printer.print_list(self.dynamic_sizes, printer.print_ssa_value)
+        printer.print_string(")")
```

Comment:
Another dumb question, is this not possible with custom-format printing/parsing?

Reply:
I don't think we handle not printing the delimiters of an optional list if it's empty. CC @math-fehr @PapyChacal is this correct?

Reply:
Soon TM, for now we can't do that :(
But @PapyChacal is working on this AFAIK